nilearn/datasets/struct.py

```diff
@@ -423,25 +424,75 @@ def fetch_oasis_vbm(n_subjects=None, dartel_version=True, data_dir=None,
         description=fdescr)
 
 
-def fetch_surf_fsaverage5(data_dir=None, url=None, resume=True, verbose=1):
-
-    """ Download Freesurfer fsaverage5 surface
+def fetch_surf_fsaverage(mesh='fsaverage5', data_dir=None):
+    """ Download a Freesurfer fsaverage surface
 
     Parameters
     ----------
-    data_dir: str, optional
+    mesh : str, optional (default='fsaverage5')
+        Which mesh to fetch.
+        'fsaverage5': the low-resolution fsaverage5 mesh (10242 nodes)
+        'fsaverage': the high-resolution fsaverage mesh (163842 nodes)
+
+    data_dir: str, optional (default=None)
         Path of the data directory. Used to force data storage in a specified
-        location. Default: None
+        location.
 
-    url: str, optional
-        Override download URL. Used for test only (or if you setup a mirror of
-        the data). Default: None
+    Returns
+    -------
+    data: sklearn.datasets.base.Bunch
+        Dictionary-like object, the interest attributes are :
+         - 'pial_left': Gifti file, left hemisphere pial surface mesh
+         - 'pial_right': Gifti file, right hemisphere pial surface mesh
+         - 'infl_left': Gifti file, left hemisphere inflated pial surface mesh
+         - 'infl_right': Gifti file, right hemisphere inflated pial
+                         surface mesh
+         - 'sulc_left': Gifti file, left hemisphere sulcal depth data
+         - 'sulc_right': Gifti file, right hemisphere sulcal depth data
 
-    resume: bool, optional (default True)
-        If True, try resuming download if possible.
+    References
+    ----------
+    Fischl et al, (1999). High-resolution intersubject averaging and a
+    coordinate system for the cortical surface. Hum Brain Mapp 8, 272-284.
+
+    """
+    meshes = {'fsaverage5': _fetch_surf_fsaverage5,
+              'fsaverage': _fetch_surf_fsaverage}
+    if mesh not in meshes:
+        raise ValueError(
+            "'mesh' should be one of {}; {!r} was provided".format(
+                list(meshes.keys()), mesh))
+    return meshes[mesh](data_dir=data_dir)
+
+
+def _fetch_surf_fsaverage(data_dir=None):
+    data_dir = _get_dataset_dir('fsaverage')
```

Comment:
None is a valid value for data_dir, it gets us the default (either in ~/nilearn_data or somewhere set with an environment variable)

Reply:
`None` as default value is certainly useful in `fetch_surf_fsaverage()`, since it is the user-facing function.
Is it also useful in `_fetch_surf_fsaverage()`, where the value of arg passed to param `data_dir` is overwritten in the next line?

    def _fetch_surf_fsaverage(data_dir=None):
        data_dir = _get_dataset_dir('fsaverage')